### PR TITLE
:sparkles: Simplify index pkg

### PR DESCRIPTION
**Pull request by mjudeikis**
## Summary

Simplify index package and wire in the ability to forward to different URLs. For now only in insecure mode in dev, as we need to wire in certificates deeper into stack. This is preparation for the experimental feature "workspace mounts"

Follow-up PR will follow: https://github.com/kcp-dev/kcp/pull/3057 

## Related issue(s)

Fixes #

## Release Notes

```release-note
Simplify index package for frontproxy
```

### CODE REVIEW

File: pkg/index/index.go [hunk 1]
```diff
@@ -92,42 +112,55 @@ func (c *State) UpsertWorkspace(shard string, ws *tenancyv1alpha1.Workspace) {
 		c.shardWorkspaceName[shard][logicalcluster.Name(ws.Spec.Cluster)] = ws.Name
 		c.shardClusterParentCluster[shard][logicalcluster.Name(ws.Spec.Cluster)] = clusterName
 	}
+
+	if mount := c.workspaceNameMount[clusterName][ws.Name]; mount != ws.Annotations[tenancyv1alpha1.ExperimentalWorkspaceMountAnnotationKey] {
```
Comment (mjudeikis) [inline pkg/index/index.go:116]: this needs to go into follow-up pr, not here.

**Comment by mjudeikis** [comment]
@sttts the first PR in mounts saga. ready to review. Closed other as it was changing too much. Now have stable branch, so gonna start moving them bit by bit

File: pkg/index/index.go [hunk 2]
```diff
@@ -28,8 +28,17 @@ import (
 
 // Index implements a mapping from logical cluster to (shard) URL.
 type Index interface {
-	Lookup(path logicalcluster.Path) (shard string, cluster, canonicalPath logicalcluster.Path, found bool)
-	LookupURL(logicalCluster logicalcluster.Path) (url string, canonicalPath logicalcluster.Path, found bool)
+	Lookup(path logicalcluster.Path) Result
```
Comment (sttts) [inline pkg/index/index.go:31]: for follow-up, maybe `Result, bool` is more canonical.

**Pull request merged**